### PR TITLE
Change hand tracking project settings and finetune `show_when_tracked`

### PR DESCRIPTION
**Pull request by BastiaanOlij**
This PR changes some of the project settings around OpenXR hand tracking.
This is response to a discussion during the XR meeting of 26 July 2024.

Two requirements were brought forward:

1. Hand tracking should be disabled by default. Enabling hand tracking has a performance cost while many games are designed around controller tracking only.
2. We should have control over which data sources are requested so I've introduced two project settings for this.

Both of these changes are **breaking** changes. They are relatively minor as users simply need to turn on the functionality that was enabled by default previously but during the XR meeting we felt that this is acceptable considering the consequences of leaving these enabled out of ignorance is worse.

Further changes in this PR include:
- Improvements in feedback on the active data source with the new type `HAND_TRACKING_SOURCE_NOT_TRACKED` added so we can make a difference between this feature being unsupported, and the feature working correctly but reporting that there is not hand tracking data being tracked.
- A minor fix in `XRNode3D` which wasn't always hiding the node in runtime when no tracking was available.

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -2928,10 +2928,22 @@
...
-			If true the hand interaction profile extension will be activated if supported by the platform.
+			If [code]true[/code] the hand interaction profile extension will be activated if supported by the platform.
 		</member>
-		<member name="xr/openxr/extensions/hand_tracking" type="bool" setter="" getter="" default="true">
-			If true we enable the hand tracking extension if available.
+		<member name="xr/openxr/extensions/hand_tracking" type="bool" setter="" getter="" default="false">
+			If [code]true[/code] we enable the hand tracking extension if available.
+
+			[b]Note:[/b] By default hand tracking will only work for data sources chosen by the XR runtime. For SteamVR this is the controller inferred data source, for most other runtimes this is the unobstructed data source. There is no way to query this. If a runtime supports the OpenXR data source extension you can use the [member xr/openxr/extensions/hand_tracking_controller_data_source] and/or [member xr/openxr/extensions/hand_tracking_unobstructed_data_source] to indicate you wish to enable these data sources. If neither is selected the data source extension is not enabled and the XR runtimes default behavior persists.
```
Comment (Kimau) [inline doc/classes/ProjectSettings.xml:2936]: So glad this infuriating bit of nonsense is documented.

**Review by Kimau — APPROVED** [review]
Looks great, nice handling of the data source issue.
Might want to call out the off by default in the headline.
As a lot of people skimming commits will only see the headline and
Hand tracking default now off is the headline change.

**Review by m4gr3d — APPROVED** [review]

File: modules/openxr/extensions/openxr_hand_tracking_extension.cpp [hunk 2]
```diff
@@ -285,23 +297,35 @@ void OpenXRHandTrackingExtension::on_process() {
...
+							XRHandTracker::HandTrackingSource source = XRHandTracker::HAND_TRACKING_SOURCE_UNKNOWN;
+							if (hand_tracking_source_ext) {
+								if (!data_source.isActive) {
+									source = XRHandTracker::HAND_TRACKING_SOURCE_NOT_TRACKED;
+								} else if (data_source.dataSource == XR_HAND_TRACKING_DATA_SOURCE_UNOBSTRUCTED_EXT) {
+									source = XRHandTracker::HAND_TRACKING_SOURCE_UNOBSTRUCTED;
+								} else if (data_source.dataSource == XR_HAND_TRACKING_DATA_SOURCE_CONTROLLER_EXT) {
+									source = XRHandTracker::HAND_TRACKING_SOURCE_CONTROLLER;
+								} else {
+									// Data source shouldn't be active, if new data sources are added to OpenXR we need to enabled them.
```
Comment (dsnopek) [inline modules/openxr/extensions/openxr_hand_tracking_extension.cpp:312]: Random tiny type-o:

```suggestion
									// Data source shouldn't be active, if new data sources are added to OpenXR we need to enable them.
```

**Review by dsnopek — APPROVED** [review]
This PR seemed to work in my testing on Meta Quest! Skimming the code, it looks good to me. :-)

The breaking changes here probably mean this PR is only for Godot 4.4, without cherry-picking to 4.3.

But this bug fix sounds like it should be cherry-picked to 4.3:

> * A minor fix in `XRNode3D` which wasn't always hiding the node in runtime when no tracking was available.

Perhaps that one should be split out to a separate PR in order to facilitate that? Or, I guess we could make a Godot 4.3 specific PR too?

File: modules/openxr/extensions/openxr_hand_tracking_extension.cpp (hunk 2 above)
Comment (akien-mga) [reply modules/openxr/extensions/openxr_hand_tracking_extension.cpp:312]: Poke @BastiaanOlij

**Comment by BastiaanOlij** [comment]
> Perhaps that one should be split out to a separate PR in order to facilitate that? Or, I guess we could make a Godot 4.3 specific PR too?

I'll make it a specific 4.3 PR, easier

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**